### ISSUE TITLE
PATCH Operations Fail with Access Policies in Directus 11.1.2+

### ISSUE THREAD

**Issue by EddiePiazza** [directus/directus#24011]
## Summary
After upgrading from Directus 11.1.1 to any newer version, PATCH operations fail for non-administrator users even when proper Access Policies are configured. This appears to be a regression in the permission handling system.

## Environment
- Last Working Version: Directus 11.1.1
- Affected Versions: 11.1.2 and above
- Authentication Method: Bearer Token
- User Type: Non-administrator users

## Description
When attempting to perform a PATCH operation with a non-administrator user who has the appropriate Access Policies configured, the system returns a FORBIDDEN error, despite the policies being correctly set up. The same operation works correctly when:
1. Using Directus version 11.1.1
2. Using an administrator account in any version

## Error Message
```json
{
  "errors": [
    {
      "message": "You don't have permissions to perform \"update\" for the field(s) \"debug\" in collection \"debug\" or it does not exist.",
      "extensions": {
        "reason": "You don't have permissions to perform \"update\" for the field(s) \"debug\" in collection \"debug\" or it does not exist.",
        "code": "FORBIDDEN"
      }
    }
  ]
}
```

#### Steps to Reproduce

Install Directus (version 11.1.2 or above)
Create a collection
Create a non-administrator user
Configure Access Policies for the user (either directly or through roles)
Authenticate using the user's Bearer Token
Attempt to perform a PATCH operation on an endpoint

#### Attempted Solutions

Applied Access Policies directly to users
Applied Access Policies to roles and assigned roles to users
Performed fresh installations to verify the issue
Verified the same configuration works correctly in version 11.1.2

#### Impact
This bug severely impacts our production environment as we rely heavily on PATCH operations for non-administrator users. The only current workaround is to grant administrator privileges, which compromises our security model.
Additional Notes

PUT operations are not affected
Administrator accounts work as expected
The issue is consistent across fresh installations
The issue appears to be specifically related to the permission checking system for PATCH operations

#### Request
Urgent assistance is needed as our production database is running on the latest version, and downgrading is not a feasible option at this point.
Reproducible Test Case
Available upon request with a minimal setup to demonstrate the issue.

### Directus Version

v11.1.2+

### Hosting Strategy

Self-Hosted (Docker Image)

### Database

Cockroach

**Comment by hanneskuettner** [comment]
Can you provide an example on how the policies are set up?

**Comment by EddiePiazza** [comment]
For sure,

1) I create a new collection called "debug", inside I create a simple string called "debug". Then I generate a new entry with null content
domain.com/items/debug
{
  "data": [
    {
      "id": 1,
      "debug": null
    }
  ]
}

2) I create a new Access Policy called "debugPolicy" and I add to this policy de collection "debug" with actions "read and update"
3) I create a new user and I add the "debugPolicy" to the user, then I generate user's token.

finally from postman or similar I just generate a PATCH to https://domain.com/items/debug/1
Json BODY:
{
    "debug": "testOK"
}

Auth Bearer Token: user's token

that will generate a "message": "You don't have permissions to perform \"update\" for the field(s) \"debug\" in collection \"debug\" or it does not exist."

access-control-allow-methods POST, GET, PATCH, OPTIONS, DELETE, PUT, HEAD, PROPFIND

So as I said, I tested over all versions from v11.1.0 to the v11.2.1
There is no problem at all until version v11.1.1 you can perfectly perform a PATCH

I downgraded it and for now I can work no problem.




> Can you provide an example on how the policies, collection and fields are set up. And also the payload you're sending with the patch.

**Comment by EddiePiazza** [comment]
Hello?

**Comment by EddiePiazza** [comment]
> I was able to replicate this issue with CockroachDB just now. It seems to be related to the way the CockroachDB driver returns numeric values as strings instead of numbers, so the check in
> 
> https://github.com/directus/directus/blob/d0c74917af60cb894e3bc1cdec6c3a9e73722879/api/src/permissions/modules/validate-access/lib/validate-item-access.ts#L54
> 
> fails.
> 
> I will submit a PR to fix this ASAP.

Hi, Is it solved? May we update?

**Comment by hanneskuettner** [comment]
This has been fixed in 11.2.2, so you're good to update!
